tests/unit/core/client/test_ConnectionManager.py

```diff
@@ -0,0 +1,235 @@
+        self.connection_manager._start()
+        self.server_port = None
+
+    def tearDown(self):
+        if self.server_port is not None:
+            self.server_port.stopListening()
+        self.connection_manager.stop()
+
+
+    @defer.inlineCallbacks
+    def test_success(self):
+        # test to see that if we setup a server, we get a connection
+        self.server = MocServerProtocolFactory(self.clock)
+        self.server_port = reactor.listenTCP(PEER_PORT, self.server, interface=LOCAL_HOST)
+        d = yield self.connection_manager.manage(schedule_next_call=False)
+        self.assertEqual(1, self.connection_manager.num_peer_connections())
+        d = yield self.connection_manager._peer_connections[self.TEST_PEER].factory.connection_was_made_deferred
+        self.assertEqual(0, self.connection_manager.num_peer_connections())
+        self.assertEqual(True, d)
+        self.assertEqual(1, self.TEST_PEER.success_count)
+        self.assertEqual(0, self.TEST_PEER.down_count)
+
+    @defer.inlineCallbacks
+    def test_bad_server(self):
+        # test to see that if we setup a server that returns an improper reply
+        # we don't get a connection
+        self.server = MocServerProtocolFactory(self.clock, is_good=False)
+        self.server_port = reactor.listenTCP(PEER_PORT, self.server, interface=LOCAL_HOST)
+        d = yield self.connection_manager.manage(schedule_next_call=False)
+        self.assertEqual(1, self.connection_manager.num_peer_connections())
+        d = yield self.connection_manager._peer_connections[self.TEST_PEER].factory.connection_was_made_deferred
+        self.assertEqual(0, self.connection_manager.num_peer_connections())
+        self.assertEqual(True, d)
+        self.assertEqual(0, self.TEST_PEER.success_count)
+        self.assertEqual(1, self.TEST_PEER.down_count)
+
+    @defer.inlineCallbacks
+    def test_non_existing_server(self):
+        # Test to see that if we don't setup a server, we don't get a connection
+        d = yield self.connection_manager.manage(schedule_next_call=False)
+        self.assertEqual(1, self.connection_manager.num_peer_connections())
+        d = yield self.connection_manager._peer_connections[self.TEST_PEER].factory.connection_was_made_deferred
+        self.assertEqual(0, self.connection_manager.num_peer_connections())
+        self.assertEqual(False, d)
+        self.assertEqual(0, self.connection_manager.num_peer_connections())
+        self.assertEqual(0, self.TEST_PEER.success_count)
+        self.assertEqual(1, self.TEST_PEER.down_count)
+
+    @defer.inlineCallbacks
+    def test_parallel_connections(self):
+        # Test to see that we make new connection for each manage call,
+        # without it waiting for the connection to complete
+        test_peer2 = Peer(LOCAL_HOST, PEER_PORT+1)
+        self.primary_request_creator.peers_to_return = [self.TEST_PEER, test_peer2]
+        d = yield self.connection_manager.manage(schedule_next_call=False)
```

Comment:
I commented on a couple of cases, but not all.  Can you go through and fix all of the `d = yield ...` instances?